ambry-tools/src/main/java/com.github.ambry/tools/perf/rest/NettyPerfClient.java

```diff
@@ -186,10 +212,15 @@ protected ClientArgs(String args[]) {
     private void validateArgs() {
       if (!SUPPORTED_REQUEST_TYPES.contains(requestType)) {
         throw new IllegalArgumentException("Unsupported request type: " + requestType);
-      } else if (requestType.equals(POST) && (postBlobTotalSize == null || postBlobTotalSize <= 0
-          || postBlobChunkSize == null || postBlobChunkSize <= 0)) {
-        throw new IllegalArgumentException(
-            "Total size to be posted and size of each chunk need to be specified with POST and have to be > 0");
+      } else if (requestType.equals(POST)) {
+        if (postBlobTotalSize == null || postBlobTotalSize <= 0 || postBlobChunkSize == null
+            || postBlobChunkSize <= 0) {
+          throw new IllegalArgumentException(
+              "Total size to be posted and size of each chunk need to be specified with POST and have to be > 0");
+        } else if (targerAccountName == null || targetContainerName == null) {
```

Comment:
service id check?

Reply:
we have defaults set for serviceId. hence null check may not make sense.